hplsql/src/main/java/org/apache/hive/hplsql/Exec.java

```diff
@@ -1822,8 +1826,12 @@ public List<Table> intoTables(ParserRuleContext ctx, List<String> names) {
    * User-defined function in a SQL query
    */
   public void execSql(String name, HplsqlParser.Expr_func_paramsContext ctx) {
-    if (execUserSql(ctx, name)) {
-      return;
+    try {
+      if (execUserSql(ctx, name)) {
+        return;
+      }
+    } catch (TException e) {
+      throw new HplValidationException(ctx, ExceptionUtils.getStackTrace(e));
```

Comment:
What is the reason not to extend `HplValidationException` with a constructor which receives `Throwable cause` to wrap the original exception?

Reply:
I will do it 👍🏼